src/components/ProfileField/template.ts

```diff
@@ -0,0 +1,23 @@
+import Handlebars from "handlebars"
+import AvatarComponent from "../../partials/AvatarComponent";
+
+Handlebars.registerPartial('AvatarComponent', AvatarComponent);
+Handlebars.registerHelper('log', (value) => { console.log(value)});
+
+
+const ProfileFieldTemplate = `
+<div class="{{styles.b-profile-field-wrapper}}">
+    <div class="{{styles.b-profile-field}}">
+        <span class="{{styles.b-profile-field-label}}">{{#if this.label}}{{this.label}}{{/if}}</span>
+        {{#if mode}}
```

Comment:
я не очень понимаю, о чем свойство mode. По коду я вижу, что это какое-то условие. Тогда будет гораздо читаемее, если это свойство будет называться, как boolean. Например так
```suggestion
        {{#if withError}}
```

Reply:
Изначально были раздельные View для авторизации и регистрации, раздельные View для просмотра профиля, редактирования профиля и смены пароля. Потом я решил все эти View объединить в AuthView и ProfileView соответственно. 

При создании какого-то комплексного View, например ProfileView я в конструкторе передаю параметр mode, котоырый и выбирает по сути, что именно будет отображено на странице. Смотрим мы свойства профиля пользователя или редактируем.

Такую же логику я инкапсулировал в компонент InputComponent, если в конструкторе мы явно передаем режим в котором вьюха сейчас, то мы выдаем инпут, что бы пользователь мог его отредактировать и отправить какую-то форму. Иначе возвращаем обычный <p>Sometext</p>

По хорошему нужно объединить InputComponent и ProfileField[Component], у меня просто не хватило на это времени, боялся не сдать второй спринт в срок.

Тут по смыслу mode можно заменить на isInput

Reply:
я понял. В таком случае, тут опять же можно сделать mode через union
```
mode: 'input' | 'auth' | 'profile' и тд
```

и в шаблоне проверять так
```
if mode === 'input'
```

причем нам компилятор TS'a сам подскажет значения mode и выдаст ошибку, если мы случайно опечатаемся